### PR TITLE
DLPX-79403 ENVIRONMENT_DISCOVER operation fails with PluginSocketNotReadyException for EBS vsdk plugin on dvp 4.0.2

### PR DESCRIPTION
**Pull request by SumoSourabh**
# Context:
When plugin developers add symlink folders to /src directory, `dvp build` does not create .pyc files in the same folder because of the missing `legacy=true` when compiling build folder.

Plugin developers can use symlink folders as they have common code which they would like to use when developing plugins for different systems like WINDOWS or UNIX.

# Problem:
Absence of .pyc files for symlinks folder after build fails the plugin imports.

# Solution:
When we copy the src folder to build folder, we copy the symlink folder as a folder to build folder. Compiling the build folder after the copy using `legacy=true` creates the .pyc files in the same folder which can be copied to zip files.

# Testing
- Build and upload a plugin with dvp==4.0.2 having symlink folder and symlink files in /src folder and /src/test1 folder (for hierarchy check). It failed because of import error.
- Build and upload the same plugin with the changes and worked.

# Notes To Reviewers:
- This change does not need code changes to delphix engine related to dvp-api.

### CODE REVIEW

**Review by vimleshmishra — APPROVED** [review]
Changes look simple and decent, and the only impact is compiling the same codebase multiple times only during the dvp build process but will limit the chance of regressions.

File: tools/src/main/python/dlpx/virtualization/_internal/commands/build.py [hunk 1]
```diff
@@ -145,6 +145,10 @@ def build(plugin_config,
     # Copy everything from the source directory into the build directory.
     file_util.clean_copy(src_dir, build_src_dir)
 
+    # compile the copied src directory. We compile here, as symlinks are also copied
+    # as folder and can be compiled without compiling the symlink source folder.
```
Comment (nhlien93) [inline tools/src/main/python/dlpx/virtualization/_internal/commands/build.py:149]: we follow the multiline comment syntax of having a `#` before and after. Also we try to do full sentences so make sure to capitalize first words for comments.

The comment is somewhat confusing to me though. or maybe i just don't understand why this was the solution. I don't think the description of the PR was updated? why are we compiling here only to install other dependencies only to compile again?

Comment (SumoSourabh) [reply tools/src/main/python/dlpx/virtualization/_internal/commands/build.py]: Updated the comment.

Also, the reason to go for this solution

- Before creating the zip file, we do compile the full build folder. But we do not use legacy=true due to which the .pyc files are not created within the same folder (created in __pycache__ folder in local cache directory) as .py files
- The above causes missing .pyc files for all the .py files present in symlinks folder.

To sort out the issue, we could have compiled the full build folder using  legacy=true which caused another issues related to protobuf version mismatch between docker engine(protobuf=3.19.0) and dvp-api(3.6.1).

So, we went via this approach where we first compile the build folder (Without any dependency in it), to make sure all pyc files are generated for .py files only in src folder.

I hope the above will help.

File: tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py [hunk 2]
```diff
@@ -66,6 +66,52 @@ def test_build_success(mock_compile_py_files, mock_relative_path, mock_install_d
...
+        mock_relative_path.assert_called()
+        mock_patch_dependencies.assert_called()
+
+        # After running build this file should now exist.
+        assert os.path.exists(artifact_file)
+
+        with open(artifact_file, 'rb') as f:
+            content = json.load(f)
+
+        # assert that source code changed because of symlink folder.
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py:107]: nit: capitol letter / full sentence with period for all comments?

**Review by nhlien93 — APPROVED** [review]
Looks great! thanks.

File: tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py (hunk 2 above)
Comment (SumoSourabh) [reply tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py]: Done.

**Pull request merged**